packages/x-charts/src/hooks/useTicksGrouped.ts

```diff
@@ -0,0 +1,215 @@
+  labelOffset: number;
+  /**
+   * In band scales, we remove some redundant ticks.
+   */
+  ignoreTick?: boolean;
+  dataIndex?: number;
+  /**
+   * The index of the group this tick belongs to. If `getGrouping` returns `[[0, 1], [2, 3]]`
+   * both ticks with value `0` and `2` will have `groupIndex: 0`, and both ticks with value `1` and `3` will have `groupIndex: 1`.
+   */
+  groupIndex?: number;
+  /**
+   * The index of the tick group inside the group. This is useful if you have multiple ticks with the same value,
+   * but they are divided by other values. Eg: `'Jan', ..., 'Dec', 'Jan'`
+   * We would have the first `'Jan'` with `syncIndex: 0` and `syncIndex: 12` for the second `'Jan'`.
+   */
+  syncIndex?: number;
+};
+
+export function useTicksGrouped(
+  options: {
+    scale: ScaleBand<any> | ScalePoint<any>;
+    valueFormatter?: AxisConfig['valueFormatter'];
+    direction: 'x' | 'y';
+    getGrouping: (value: any, dataIndex: number) => any[];
+  } & Pick<TickParams, 'tickNumber' | 'tickInterval' | 'tickPlacement' | 'tickLabelPlacement'>,
+): GroupedTickItemType[] {
+  const {
+    scale,
+    tickInterval,
+    tickLabelPlacement = 'middle',
+    tickPlacement = 'extremities',
+    getGrouping,
+  } = options;
+
+  return React.useMemo(() => {
+    const domain = scale.domain();
+    const filteredDomain =
+      (typeof tickInterval === 'function' && domain.filter(tickInterval)) ||
+      (typeof tickInterval === 'object' && tickInterval) ||
+      domain;
+
+    if (scale.bandwidth() > 0) {
+      // scale type = 'band'
+      const { entries, maxGroupIndex } = mapToGrouping(
+        filteredDomain,
+        getGrouping,
+        tickPlacement,
+        tickLabelPlacement,
+        scale,
+      );
+
+      if (entries[0]) {
+        entries[0].ignoreTick = true;
+      }
+
+      return [
+        {
+          formattedValue: undefined,
+          offset: scale.range()[0],
+          labelOffset: 0,
+          groupIndex: maxGroupIndex,
+        },
+
+        ...entries,
+
+        // Last tick
+        {
+          formattedValue: undefined,
+          offset: scale.range()[1],
+          labelOffset: 0,
+          groupIndex: maxGroupIndex,
+        },
+      ];
+    }
+
+    // scale type = 'point'
+    const { entries } = mapToGrouping(
+      filteredDomain,
+      getGrouping,
+      tickPlacement ?? 'extremities',
+      tickLabelPlacement ?? 'middle',
+      scale,
+    );
+    return entries;
+  }, [scale, tickInterval, getGrouping, tickPlacement, tickLabelPlacement]);
+}
+
+function mapToGrouping(
+  tickValues: any[],
+  getGrouping: (value: any, dataIndex: number) => any[],
+  tickPlacement: Exclude<TickParams['tickPlacement'], undefined>,
+  tickLabelPlacement: Exclude<TickParams['tickLabelPlacement'], undefined>,
+  scale: D3Scale,
+) {
+  // Step 1: Create all tick items with their group information
+  const allTickItems: Omit<Required<GroupedTickItemType>, 'syncIndex'>[] = [];
+
+  let maxGroupIndex = 0;
```

Comment:
That probably come from the initial implementation. Now you know that `maxGroupIndex` is the `groups.length-1`